components/knowledge/Notion.tsx

```diff
@@ -0,0 +1,204 @@
+    try {
+      const isConfigured = await isNotionConfigured();
+      await runNotionSync(isConfigured);
+      setNotionConfigured(isConfigured);
+      setNotionFiles(await getNotionFiles());
+      setSyncError('');
+    } catch (e) {
+      setSyncError((e as Error).toString());
+    } finally {
+      setIsSyncing(false);
+    }
+  };
+
+  const handleSelectedFileChange = (selected: any) => {
+    if (selected === 'all') {
+      setSelectedFile(Array.from(notionFiles.keys()));
+    } else {
+      setSelectedFile(Array.from(selected as Set<string>));
+    }
+  };
+
+  return (
+    <Modal
+      size="4xl"
+      isOpen={isOpen}
+      onClose={onClose}
+      hideCloseButton={true}
+      scrollBehavior="inside"
+      className="min-h-[40vh] max-h-[80vh]"
+    >
+      <ModalContent>
+        <ModalHeader>
+          <div className="w-full flex items-center justify-between">
+            <div className="flex flex-row items-center">
+              <Image className="h-6 w-6" src="notion.svg" alt="Notion Icon" />
+              <p className="ml-2">Notion</p>
+            </div>
+
+            <div className="flex items-center justify-end p-2">
+              <Button
+                isLoading={isSyncing}
+                size="sm"
+                color="primary"
+                startContent={!isSyncing && <IoMdRefresh />}
+                onClick={syncNotion}
+              >
+                {!isSyncing && 'Sync'}
+              </Button>
+              {syncError && (
+                <p className="text-sm text-red-500 ml-2">{syncError}</p>
+              )}
+            </div>
+          </div>
+        </ModalHeader>
+        <ModalBody>
+          {notionConfigured && notionFiles.size > 0 ? (
+            <div className="flex flex-col gap-1">
+              <Table
+                selectionMode="multiple"
+                selectionBehavior="toggle"
+                aria-label="notion-files"
+                selectedKeys={selectedFile}
+                onSelectionChange={(selected) => {
+                  handleSelectedFileChange(selected);
+                }}
+              >
+                <TableHeader>
+                  <TableColumn>NAME</TableColumn>
+                  <TableColumn>Link</TableColumn>
+                </TableHeader>
+                <TableBody>
+                  {Array.from(notionFiles.entries()).map(([key, value]) => (
+                    <TableRow key={key}>
+                      <TableCell>
+                        <div className="flex flex-col">
+                          <p>{value.fileName}</p>
+                        </div>
+                      </TableCell>
+                      <TableCell>
+                        <Button
+                          isIconOnly
+                          color="primary"
+                          variant="flat"
+                          startContent={<CiShare1 />}
+                          onClick={() => {
+                            window.open(value.url);
```

Comment:
why use `window.open` here?

Reply:
Any good alternative?

Reply:
Never seen it used in React before, and wanted to see what the intention here was before I gave an opinion.

Reply:
Usually for rerouting users use `useRouter` (client + import from `next/navigation`) or `redirect` (server)

But for opening external links in the users default browser I'd go with `<Link isExternal />` imported from `@nextui-org/react`

Not sure if these fit your use case though so feel free to chime in

Reply:
Yeah I will change it to use Link. I have no experience though.

Reply:
Looks Great now!